src/pages/categories/Categories.tsx

```diff
@@ -1,22 +1,85 @@
+import { useCallback, useMemo } from 'react'
 import { useTranslation } from 'react-i18next'
+
 import { useModalContext } from '~/context/modal-context'
+import useBreakpoints from '~/hooks/use-breakpoints'
+import useLoadMore from '~/hooks/use-load-more'
 
+import palette from '~/styles/app-theme/app.pallete'
+import { itemsLoadLimit } from '~/constants'
+import { authRoutes } from '~/router/constants/authRoutes'
+import { CategoryInterface } from '~/types'
+import { categoryService } from '~/services/category-service'
+import { getScreenBasedLimit } from '~/utils/helper-functions'
 import CreateSubjectModal from '~/containers/find-offer/create-new-subject/CreateNewSubject'
 import NotFoundResults from '~/components/not-found-results/NotFoundResults'
 import OfferRequestBlock from '~/containers/find-offer/offer-request-block/OfferRequestBlock'
 import PageWrapper from '~/components/page-wrapper/PageWrapper'
+import CardWithLink from '~/components/card-with-link/CardWithLink'
+import CardsList from '~/components/cards-list/CardsList'
+import { categoryIcons } from '~/constants/icon-map'
 
-const categories = null
+type CategoryIconKey = keyof typeof categoryIcons
 
 const Categories = () => {
   const { t } = useTranslation()
+  const breakpoints = useBreakpoints()
   const { openModal } = useModalContext()
+  const cardsLimit = getScreenBasedLimit(breakpoints, itemsLoadLimit)
+
+  const getCategories = useCallback(
+    (data?: Pick<CategoryInterface, 'name'>) =>
+      categoryService.getCategories(data),
+    []
+  )
```

Comment:
we don't need  useCallback here because categoryService and function will not be regenerated on every render.
Why don't we just use  categoryService.getCategories where this function is needed?

Reply:
updated